api/alarms/index.js

```diff
@@ -0,0 +1,182 @@
+let display = document.querySelector('.alarm-display');
+let log = document.querySelector('.alarm-log');
+let form = document.querySelector('.create-alarm');
+let clearButton = document.getElementById('clear-display');
+let refreshButton = document.getElementById('refresh-display');
+
+// DOM event bindings
+
+//// Alarm display buttons
+
+clearButton.addEventListener('click', () => manager.cancelAllAlarms());
+refreshButton.addEventListener('click', () => manager.refreshDisplay());
+
+//// New alarm form
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+  let formData = new FormData(form);
+  let data = Object.fromEntries(formData);
+
+  // Extract form values
+  let name = data['alarm-name'];
+  let delay = Number.parseFloat(data['time-value']);
+  let delayFormat = data['time-format'];
+  let period = Number.parseFloat(data['period']);
+
+  // Prepare alarm info for creation call
+  let alarmInfo = {};
+
+  if (delayFormat === 'ms') {
+    // Specified in milliseconds, use `when` property
+    alarmInfo.when = Date.now() + delay;
+  } else if (delayFormat === 'min') {
+    // specified in minutes, use `delayInMinutes` property
+    alarmInfo.delayInMinutes = delay;
+  }
+
+  if (period) {
+    alarmInfo.periodInMinutes = period;
+  }
+
+  // Create the alarm – this uses the same signature as chrome.alarms.create
+  manager.createAlarm(name, alarmInfo);
+});
+
+class AlarmManager {
+  constructor({display, log} = {}) {
+    if (display) {
+      this.displayElement = display;
+    }
+    if (log) {
+      this.logElement = log;
+    }
+
+    this.logMessage('Manager: initializing demo');
+
+    this.displayElement.addEventListener('click', (event) => this.handleCancelAlarm(event));
```

Comment:
Another way to do this is to define `handleCancelAlarm` as a property of this class, e.g.:

```js
  // instead of
  async handleCancelAlarm(event) { ... }

  // do this
  handleCancelAlarm = (event) => {};
```

That way it's already bound to `this` and your `addEventListener` calls are simpler.

Reply:
Ack.